clang/lib/Sema/SemaOpenACC.cpp

```diff
@@ -82,22 +119,22 @@ bool SemaOpenACC::ActOnStartStmtDirective(OpenACCDirectiveKind K,
   return diagnoseConstructAppertainment(*this, K, StartLoc, /*IsStmt=*/true);
 }
 
-StmtResult SemaOpenACC::ActOnEndStmtDirective(OpenACCDirectiveKind K,
-                                              SourceLocation StartLoc,
-                                              SourceLocation EndLoc,
-                                              StmtResult AssocStmt) {
+StmtResult SemaOpenACC::ActOnEndStmtDirective(
+    OpenACCDirectiveKind K, SourceLocation StartLoc, SourceLocation EndLoc,
+    MutableArrayRef<OpenACCClause *> Clauses, StmtResult AssocStmt) {
```

Comment:
```suggestion
    ArrayRef<OpenACCClause *> Clauses, StmtResult AssocStmt) {
```

Reply:
The result of these changes is that the 'DestroyUnusedClauses' can no longer set the values to `nullptr`, and also 'deleting' stuff that were passed by an ArrayRef.  

This is actually ALSO causing me to think we should be passing this as an llvm::SmallVector (and moving it into here!) to make it clear that this function is stealing the 'ownership' of these things.  WDYT?

Reply:
Better to use OwningArrayRef, if changes do not require resizing, I assume

Reply:
I think I'm trying too much to gild a lily here, I think your suggestion of removing the 'destroy' function + just being ArrayRef is acceptable, so the next patch will do that.